internal/backend/dispatcher/eventsworkflow.go

```diff
@@ -24,6 +32,62 @@ func (d *dispatcher) createEventRecord(ctx context.Context, eventID sdktypes.Eve
 	return nil
 }
 
+func (d *dispatcher) resolveEnv(ctx context.Context, env string) (envID sdktypes.EnvID, err error) {
+	if env == "" {
+		return sdktypes.InvalidEnvID, nil
+	}
+
+	parts := strings.Split(env, "/")
+	switch len(parts) {
+	case 1:
```

Comment:
since there are only 2 cases, I would suggest to use simple `if` and remove nesting for the second part